test values: ./runTests.py -x --ndim 3
{'ndim': 3, 'flat': None, 'seed': None, 'new': False, 'flatness': (False, False, False), 'x': True}
...............................................................................................F...F...........FF.................E....................E................................F..........F.F.........F...F...F...F...........FF.......
======================================================================
ERROR: testDiv (unitTests.productTester)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/suki/Tutorials/personal-projects/mvar/unitTests.py", line 260, in testDiv
    self.assertTrue( self.A/self.B == self.A*(self.B**(-1)) )
  File "/home/suki/Tutorials/personal-projects/mvar/decorate/automath.py", line 71, in __div__
    return self*other**(-1)
  File "<string>", line 2, in __mul__
  File "/home/suki/Tutorials/personal-projects/mvar/decorate/__init__.py", line 50, in F
    return decorated(*before(*args),**kwargs)
  File "<string>", line 2, in __mul__
  File "/home/suki/Tutorials/personal-projects/mvar/decorate/__init__.py", line 140, in caller
    return found(*args,**kwargs)
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 1780, in _mvarMul
    result = (self*mvarA*mvarB+mvar*selfA*selfB)
  File "<string>", line 2, in __mul__
  File "/home/suki/Tutorials/personal-projects/mvar/decorate/__init__.py", line 50, in F
    return decorated(*before(*args),**kwargs)
  File "<string>", line 2, in __mul__
  File "/home/suki/Tutorials/personal-projects/mvar/decorate/__init__.py", line 140, in caller
    return found(*args,**kwargs)
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 1754, in _matrixMul
    vectors=self.vectors*matrix,
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 363, in __init__
    self.copy(self.square())
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 519, in square
    var=result.var,
  File "/home/suki/Tutorials/personal-projects/mvar/square.py", line 71, in square
    (var,vectors) = _subSquare(vectors,var)
  File "/home/suki/Tutorials/personal-projects/mvar/square.py", line 124, in _subSquare
    (val,vec)=eig(cov)
  File "/usr/lib/pymodules/python2.7/numpy/linalg/linalg.py", line 1187, in eigh
    raise LinAlgError, 'Eigenvalues did not converge'
LinAlgError: Eigenvalues did not converge

======================================================================
ERROR: testRight (unitTests.commuteTester)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/suki/Tutorials/personal-projects/mvar/unitTests.py", line 32, in testRight
    self.assertTrue( self.A/self.B == (self.B**-1)*self.A )
  File "/home/suki/Tutorials/personal-projects/mvar/decorate/automath.py", line 71, in __div__
    return self*other**(-1)
  File "<string>", line 2, in __mul__
  File "/home/suki/Tutorials/personal-projects/mvar/decorate/__init__.py", line 50, in F
    return decorated(*before(*args),**kwargs)
  File "<string>", line 2, in __mul__
  File "/home/suki/Tutorials/personal-projects/mvar/decorate/__init__.py", line 140, in caller
    return found(*args,**kwargs)
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 1780, in _mvarMul
    result = (self*mvarA*mvarB+mvar*selfA*selfB)
  File "<string>", line 2, in __mul__
  File "/home/suki/Tutorials/personal-projects/mvar/decorate/__init__.py", line 50, in F
    return decorated(*before(*args),**kwargs)
  File "<string>", line 2, in __mul__
  File "/home/suki/Tutorials/personal-projects/mvar/decorate/__init__.py", line 140, in caller
    return found(*args,**kwargs)
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 1754, in _matrixMul
    vectors=self.vectors*matrix,
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 363, in __init__
    self.copy(self.square())
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 519, in square
    var=result.var,
  File "/home/suki/Tutorials/personal-projects/mvar/square.py", line 71, in square
    (var,vectors) = _subSquare(vectors,var)
  File "/home/suki/Tutorials/personal-projects/mvar/square.py", line 124, in _subSquare
    (val,vec)=eig(cov)
  File "/usr/lib/pymodules/python2.7/numpy/linalg/linalg.py", line 1187, in eigh
    raise LinAlgError, 'Eigenvalues did not converge'
LinAlgError: Eigenvalues did not converge

======================================================================
FAIL: given (__init__.Mvar)
Doctest: __init__.Mvar.given
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/doctest.py", line 2166, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for __init__.Mvar.given
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 929, in given

----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 986, in __init__.Mvar.given
Failed example:
    !! this is one step from a reverse chain ?
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.given[21]>", line 1
        !! this is one step from a reverse chain ?
        ^
    SyntaxError: invalid syntax


======================================================================
FAIL: marginal (__init__.Mvar)
Doctest: __init__.Mvar.marginal
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/doctest.py", line 2166, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for __init__.Mvar.marginal
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 1040, in marginal

----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 1048, in __init__.Mvar.marginal
Failed example:
    assert reduce(operator.and_,M) == A.diag()
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.marginal[3]>", line 1, in <module>
        assert reduce(operator.and_,M) == A.diag()
      File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 1427, in __and__
        Dmean = Plane.__and__(self,other).mean
      File "/home/suki/Tutorials/personal-projects/mvar/plane.py", line 112, in __and__
        Nself=self.getNull()
      File "/home/suki/Tutorials/personal-projects/mvar/plane.py", line 105, in getNull
        return getNull(self.vectors)
      File "/home/suki/Tutorials/personal-projects/mvar/plane.py", line 20, in getNull
        var,vectors = square(vectors)
      File "/home/suki/Tutorials/personal-projects/mvar/square.py", line 71, in square
        (var,vectors) = _subSquare(vectors,var)
      File "/home/suki/Tutorials/personal-projects/mvar/square.py", line 124, in _subSquare
        (val,vec)=eig(cov)
      File "/usr/lib/pymodules/python2.7/numpy/linalg/linalg.py", line 1187, in eigh
        raise LinAlgError, 'Eigenvalues did not converge'
    LinAlgError: Eigenvalues did not converge


======================================================================
FAIL: transform (__init__.Mvar)
Doctest: __init__.Mvar.transform
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/doctest.py", line 2166, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for __init__.Mvar.transform
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 612, in transform

----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 624, in __init__.Mvar.transform
Failed example:
    assert (A**numpy.real(K1)).transform() == A.transform(numpy.real(K1)) 
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.transform[7]>", line 1, in <module>
        assert (A**numpy.real(K1)).transform() == A.transform(numpy.real(K1))
    AssertionError


======================================================================
FAIL: width (__init__.Mvar)
Doctest: __init__.Mvar.width
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/doctest.py", line 2166, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for __init__.Mvar.width
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 789, in width

----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 794, in __init__.Mvar.width
Failed example:
    assert Matrix([A[n].var[0] for n in range(A.ndim)]) == A.width()**2
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.width[0]>", line 1, in <module>
        assert Matrix([A[n].var[0] for n in range(A.ndim)]) == A.width()**2
    IndexError: index out of bounds
----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 796, in __init__.Mvar.width
Failed example:
    assert Matrix(A.corr.diagonal()) == Matrix.ones
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.width[1]>", line 1, in <module>
        assert Matrix(A.corr.diagonal()) == Matrix.ones
    AssertionError
----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 799, in __init__.Mvar.width
Failed example:
    assert norm.corr == norm.cov
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.width[3]>", line 1, in <module>
        assert norm.corr == norm.cov
      File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 552, in fget
        return (self/self.width()).cov
      File "/home/suki/Tutorials/personal-projects/mvar/decorate/automath.py", line 71, in __div__
        return self*other**(-1)
      File "<string>", line 2, in __mul__
      File "/home/suki/Tutorials/personal-projects/mvar/decorate/__init__.py", line 50, in F
        return decorated(*before(*args),**kwargs)
      File "<string>", line 2, in __mul__
      File "/home/suki/Tutorials/personal-projects/mvar/decorate/__init__.py", line 140, in caller
        return found(*args,**kwargs)
      File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 1803, in __vectorMul__
        var=self.var,
      File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 360, in __init__
        ),'real numbers only'
    AssertionError: real numbers only
----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 800, in __init__.Mvar.width
Failed example:
    assert Matrix([norm[n].var[0] for n in range(norm.ndim)]) == Matrix.ones
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.width[4]>", line 1, in <module>
        assert Matrix([norm[n].var[0] for n in range(norm.ndim)]) == Matrix.ones
    AssertionError


======================================================================
FAIL: KLdiv (__init__.Mvar)
Doctest: __init__.Mvar.KLdiv
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/doctest.py", line 2166, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for __init__.Mvar.KLdiv
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 2054, in KLdiv

----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 2073, in __init__.Mvar.KLdiv
Failed example:
    assert Matrix(A.KLdiv(B)) == (A*m).KLdiv(B*m)
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.KLdiv[5]>", line 1, in <module>
        assert Matrix(A.KLdiv(B)) == (A*m).KLdiv(B*m)
    AssertionError
----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 2099, in __init__.Mvar.KLdiv
Failed example:
    assert KL.dist2() < (Z**2)
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.KLdiv[19]>", line 1, in <module>
        assert KL.dist2() < (Z**2)
    AssertionError
----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 2106, in __init__.Mvar.KLdiv
Failed example:
    assert KL.dist2() < (Z**2)
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.KLdiv[25]>", line 1, in <module>
        assert KL.dist2() < (Z**2)
    AssertionError


======================================================================
FAIL: __mul__ (__init__.Mvar)
Doctest: __init__.Mvar.__mul__
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/doctest.py", line 2166, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for __init__.Mvar.__mul__
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 28, in __mul__

----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 119, in __init__.Mvar.__mul__
Failed example:
    assert A/B == A*(B**(-1))
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.__mul__[32]>", line 1, in <module>
        assert A/B == A*(B**(-1))
      File "/home/suki/Tutorials/personal-projects/mvar/decorate/automath.py", line 71, in __div__
        return self*other**(-1)
      File "<string>", line 2, in __mul__
      File "/home/suki/Tutorials/personal-projects/mvar/decorate/__init__.py", line 50, in F
        return decorated(*before(*args),**kwargs)
      File "<string>", line 2, in __mul__
      File "/home/suki/Tutorials/personal-projects/mvar/decorate/__init__.py", line 140, in caller
        return found(*args,**kwargs)
      File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 1780, in _mvarMul
        result = (self*mvarA*mvarB+mvar*selfA*selfB)
      File "<string>", line 2, in __mul__
      File "/home/suki/Tutorials/personal-projects/mvar/decorate/__init__.py", line 50, in F
        return decorated(*before(*args),**kwargs)
      File "<string>", line 2, in __mul__
      File "/home/suki/Tutorials/personal-projects/mvar/decorate/__init__.py", line 140, in caller
        return found(*args,**kwargs)
      File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 1754, in _matrixMul
        vectors=self.vectors*matrix,
      File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 363, in __init__
        self.copy(self.square())
      File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 519, in square
        var=result.var,
      File "/home/suki/Tutorials/personal-projects/mvar/square.py", line 71, in square
        (var,vectors) = _subSquare(vectors,var)
      File "/home/suki/Tutorials/personal-projects/mvar/square.py", line 124, in _subSquare
        (val,vec)=eig(cov)
      File "/usr/lib/pymodules/python2.7/numpy/linalg/linalg.py", line 1187, in eigh
        raise LinAlgError, 'Eigenvalues did not converge'
    LinAlgError: Eigenvalues did not converge


======================================================================
FAIL: __rmul__ (__init__.Mvar)
Doctest: __init__.Mvar.__rmul__
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/doctest.py", line 2166, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for __init__.Mvar.__rmul__
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 28, in __rmul__

----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 65, in __init__.Mvar.__rmul__
Failed example:
    assert A/B == A*(B**(-1))
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.__rmul__[7]>", line 1, in <module>
        assert A/B == A*(B**(-1))
      File "/home/suki/Tutorials/personal-projects/mvar/decorate/automath.py", line 71, in __div__
        return self*other**(-1)
      File "<string>", line 2, in __mul__
      File "/home/suki/Tutorials/personal-projects/mvar/decorate/__init__.py", line 50, in F
        return decorated(*before(*args),**kwargs)
      File "<string>", line 2, in __mul__
      File "/home/suki/Tutorials/personal-projects/mvar/decorate/__init__.py", line 140, in caller
        return found(*args,**kwargs)
      File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 1780, in _mvarMul
        result = (self*mvarA*mvarB+mvar*selfA*selfB)
      File "<string>", line 2, in __mul__
      File "/home/suki/Tutorials/personal-projects/mvar/decorate/__init__.py", line 50, in F
        return decorated(*before(*args),**kwargs)
      File "<string>", line 2, in __mul__
      File "/home/suki/Tutorials/personal-projects/mvar/decorate/__init__.py", line 140, in caller
        return found(*args,**kwargs)
      File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 1754, in _matrixMul
        vectors=self.vectors*matrix,
      File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 363, in __init__
        self.copy(self.square())
      File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 519, in square
        var=result.var,
      File "/home/suki/Tutorials/personal-projects/mvar/square.py", line 71, in square
        (var,vectors) = _subSquare(vectors,var)
      File "/home/suki/Tutorials/personal-projects/mvar/square.py", line 124, in _subSquare
        (val,vec)=eig(cov)
      File "/usr/lib/pymodules/python2.7/numpy/linalg/linalg.py", line 1187, in eigh
        raise LinAlgError, 'Eigenvalues did not converge'
    LinAlgError: Eigenvalues did not converge


======================================================================
FAIL: density (__init__.Mvar)
Doctest: __init__.Mvar.density
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/doctest.py", line 2166, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for __init__.Mvar.density
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 1944, in density

----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 1960, in __init__.Mvar.density
Failed example:
    assert Matrix(0) == ratio.var()
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.density[7]>", line 1, in <module>
        assert Matrix(0) == ratio.var()
    AssertionError


======================================================================
FAIL: entropy (__init__.Mvar)
Doctest: __init__.Mvar.entropy
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/doctest.py", line 2166, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for __init__.Mvar.entropy
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 1965, in entropy

----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 2027, in __init__.Mvar.entropy
Failed example:
    assert deltas.dist2() < (Z**2)  
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.entropy[23]>", line 1, in <module>
        assert deltas.dist2() < (Z**2)
    AssertionError


======================================================================
FAIL: given (__init__.Mvar)
Doctest: __init__.Mvar.given
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/doctest.py", line 2166, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for __init__.Mvar.given
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 929, in given

----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 986, in __init__.Mvar.given
Failed example:
    !! this is one step from a reverse chain ?
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.given[21]>", line 1
        !! this is one step from a reverse chain ?
        ^
    SyntaxError: invalid syntax


======================================================================
FAIL: marginal (__init__.Mvar)
Doctest: __init__.Mvar.marginal
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/doctest.py", line 2166, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for __init__.Mvar.marginal
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 1040, in marginal

----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 1048, in __init__.Mvar.marginal
Failed example:
    assert reduce(operator.and_,M) == A.diag()
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.marginal[3]>", line 1, in <module>
        assert reduce(operator.and_,M) == A.diag()
      File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 1427, in __and__
        Dmean = Plane.__and__(self,other).mean
      File "/home/suki/Tutorials/personal-projects/mvar/plane.py", line 112, in __and__
        Nself=self.getNull()
      File "/home/suki/Tutorials/personal-projects/mvar/plane.py", line 105, in getNull
        return getNull(self.vectors)
      File "/home/suki/Tutorials/personal-projects/mvar/plane.py", line 20, in getNull
        var,vectors = square(vectors)
      File "/home/suki/Tutorials/personal-projects/mvar/square.py", line 71, in square
        (var,vectors) = _subSquare(vectors,var)
      File "/home/suki/Tutorials/personal-projects/mvar/square.py", line 124, in _subSquare
        (val,vec)=eig(cov)
      File "/usr/lib/pymodules/python2.7/numpy/linalg/linalg.py", line 1187, in eigh
        raise LinAlgError, 'Eigenvalues did not converge'
    LinAlgError: Eigenvalues did not converge


======================================================================
FAIL: transform (__init__.Mvar)
Doctest: __init__.Mvar.transform
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/doctest.py", line 2166, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for __init__.Mvar.transform
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 612, in transform

----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 621, in __init__.Mvar.transform
Failed example:
    assert A.transform(N)== (A**N).transform()
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.transform[5]>", line 1, in <module>
        assert A.transform(N)== (A**N).transform()
    AssertionError
----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 622, in __init__.Mvar.transform
Failed example:
    assert A.transform(N)== A.transform()**N  
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.transform[6]>", line 1, in <module>
        assert A.transform(N)== A.transform()**N
      File "/usr/lib/pymodules/python2.7/numpy/matrixlib/defmatrix.py", line 343, in __pow__
        return matrix_power(self, other)
      File "/usr/lib/pymodules/python2.7/numpy/matrixlib/defmatrix.py", line 171, in matrix_power
        M = inv(M)
      File "/usr/lib/pymodules/python2.7/numpy/linalg/linalg.py", line 445, in inv
        return wrap(solve(a, identity(a.shape[0], dtype=a.dtype)))
      File "/usr/lib/pymodules/python2.7/numpy/linalg/linalg.py", line 328, in solve
        raise LinAlgError, 'Singular matrix'
    LinAlgError: Singular matrix


======================================================================
FAIL: width (__init__.Mvar)
Doctest: __init__.Mvar.width
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/doctest.py", line 2166, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for __init__.Mvar.width
  File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 789, in width

----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 794, in __init__.Mvar.width
Failed example:
    assert Matrix([A[n].var[0] for n in range(A.ndim)]) == A.width()**2
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.width[0]>", line 1, in <module>
        assert Matrix([A[n].var[0] for n in range(A.ndim)]) == A.width()**2
    IndexError: index out of bounds
----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 796, in __init__.Mvar.width
Failed example:
    assert Matrix(A.corr.diagonal()) == Matrix.ones
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.width[1]>", line 1, in <module>
        assert Matrix(A.corr.diagonal()) == Matrix.ones
    AssertionError
----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 799, in __init__.Mvar.width
Failed example:
    assert norm.corr == norm.cov
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.width[3]>", line 1, in <module>
        assert norm.corr == norm.cov
      File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 552, in fget
        return (self/self.width()).cov
      File "/home/suki/Tutorials/personal-projects/mvar/decorate/automath.py", line 71, in __div__
        return self*other**(-1)
      File "<string>", line 2, in __mul__
      File "/home/suki/Tutorials/personal-projects/mvar/decorate/__init__.py", line 50, in F
        return decorated(*before(*args),**kwargs)
      File "<string>", line 2, in __mul__
      File "/home/suki/Tutorials/personal-projects/mvar/decorate/__init__.py", line 140, in caller
        return found(*args,**kwargs)
      File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 1803, in __vectorMul__
        var=self.var,
      File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 360, in __init__
        ),'real numbers only'
    AssertionError: real numbers only
----------------------------------------------------------------------
File "/home/suki/Tutorials/personal-projects/mvar/__init__.py", line 800, in __init__.Mvar.width
Failed example:
    assert Matrix([norm[n].var[0] for n in range(norm.ndim)]) == Matrix.ones
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __init__.Mvar.width[4]>", line 1, in <module>
        assert Matrix([norm[n].var[0] for n in range(norm.ndim)]) == Matrix.ones
    AssertionError


----------------------------------------------------------------------
Ran 240 tests in 2.517s

FAILED (failures=13, errors=2)
